server/apps/immich/src/modules/immich-auth/immich-jwt.service.ts

```diff
@@ -0,0 +1,95 @@
+import {BadRequestException, Injectable, InternalServerErrorException, Logger} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { JwtPayloadDto } from '../../api-v1/auth/dto/jwt-payload.dto';
+import { jwtSecret } from '../../constants/jwt.constant';
+import * as bcrypt from 'bcrypt';
+import {UserEntity} from "@app/database/entities/user.entity";
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { ImmichAuthService } from './immich-auth.service';
+import {mapUser} from "../../api-v1/user/response-dto/user-response.dto";
+
+@Injectable()
+export class ImmichJwtService {
+  constructor(
+    @InjectRepository(UserEntity)
+    private userRepository: Repository<UserEntity>,
+    private jwtService: JwtService,
+    private immichAuthService: ImmichAuthService,
+  ) {}
+
+  public async generateToken(payload: JwtPayloadDto) {
+    return this.jwtService.sign({
+      ...payload,
+    });
+  }
+
+  public async validateToken(accessToken: string) {
+    try {
+      const payload = await this.jwtService.verify(accessToken, { secret: jwtSecret });
+      return {
+        userId: payload['userId'],
+        status: true,
+      };
+    } catch (e) {
+      Logger.error('Error validating token from websocket request', 'ValidateWebsocketToken');
+      return {
+        userId: null,
+        status: false,
+      };
+    }
+  }
+
+  private async validateLocalUser(email: string, password: string): Promise<UserEntity | undefined> {
+    const user = await this.userRepository.findOne(
+        { where:
+                  { email: email },
+                 select: [
+                      'id',
+                      'email',
+                      'password',
+                      'salt',
+                      'firstName',
+                      'lastName',
+                      'isAdmin',
+                      'profileImagePath',
+                      'shouldChangePassword',
+                      'isLocalUser',
+                    ],
+                  },
+    );
+
+    if (!user || !user.isLocalUser) throw new BadRequestException('Incorrect email or password');
```

Comment:
This is just cosmetic but I prefer to have curly braces for conditional statements. Probably I will set the rules up in tslint somewhere :P

Reply:
Fixed, I also added the eslint rule to force the use of curly braces